can/interfaces/etas/__init__.py

```diff
@@ -0,0 +1,409 @@
+                f.flagsMask = OCI_CAN_MSG_FLAG_SELFRECEPTION
+            if filter.get("extended"):
+                f.flagsValue |= OCI_CAN_MSG_FLAG_EXTENDED
+                f.flagsMask |= OCI_CAN_MSG_FLAG_EXTENDED
+            self._oci_filters[i].contents = f
+
+        ec = OCI_AddCANFrameFilterEx(
+            self.rxQueue, self._oci_filters, len(self._oci_filters)
+        )
+        if ec != 0x0:
+            raise can.exceptions.CanOperationError(
+                f"OCI_AddCANFrameFilterEx failed with error 0x{ec:X}"
+            )
+
+    def flush_tx_buffer(self) -> None:
+        ec = OCI_ResetQueue(self.txQueue)
+        if ec != 0x0:
+            raise can.exceptions.CanOperationError(
+                f"OCI_ResetQueue failed with error 0x{ec:X}"
+            )
+
+    def shutdown(self) -> None:
+        # Cleanup TX
+        if self.txQueue:
+            ec = OCI_DestroyCANTxQueue(self.txQueue)
+            if ec != 0x0:
+                raise can.exceptions.CanOperationError(
+                    f"OCI_DestroyCANTxQueue failed with error 0x{ec:X}"
+                )
+            self.txQueue = None
+
+        # Cleanup RX
+        if self.rxQueue:
+            ec = OCI_DestroyCANRxQueue(self.rxQueue)
+            if ec != 0x0:
+                raise can.exceptions.CanOperationError(
+                    f"OCI_DestroyCANRxQueue failed with error 0x{ec:X}"
+                )
+            self.rxQueue = None
+
+        # Cleanup common
+        if self.ctrl:
+            ec = OCI_CloseCANController(self.ctrl)
+            if ec != 0x0:
+                raise can.exceptions.CanOperationError(
+                    f"OCI_CloseCANController failed with error 0x{ec:X}"
+                )
+            ec = OCI_DestroyCANController(self.ctrl)
+            if ec != 0x0:
+                raise can.exceptions.CanOperationError(
+                    f"OCI_DestroyCANController failed with error 0x{ec:X}"
+                )
+            self.ctrl = None
+
+        if self.tree:
+            ec = CSI_DestroyProtocolTree(self.tree)
+            if ec != 0x0:
+                raise can.exceptions.CanOperationError(
+                    f"CSI_DestroyProtocolTree failed with error 0x{ec:X}"
+                )
+            self.tree = None
+
+    @property
+    def state(self) -> can.BusState:
+        status = OCI_CANControllerStatus()
+        ec = OCI_GetCANControllerStatus(self.ctrl, ctypes.byref(status))
+        if ec != 0x0:
+            raise can.exceptions.CanOperationError(
+                f"OCI_GetCANControllerStatus failed with error 0x{ec:X}"
+            )
+        if status.stateCode & OCI_CAN_STATE_ACTIVE:
+            return can.BusState.ACTIVE
+        elif status.stateCode & OCI_CAN_STATE_PASSIVE:
+            return can.BusState.PASSIVE
+
+    @state.setter
+    def state(self, new_state: can.BusState) -> None:
+        # if new_state == can.BusState.ACTIVE:
```

Comment:
If there is a problem with the disabled code, one should probably add a brief comment on what's up.